src/node/interfaces.cpp

```diff
@@ -973,7 +973,9 @@ class MinerImpl : public Mining
         {
             WAIT_LOCK(notifications().m_tip_block_mutex, lock);
             notifications().m_tip_block_cv.wait_for(lock, timeout, [&]() EXCLUSIVE_LOCKS_REQUIRED(notifications().m_tip_block_mutex) {
-                return (notifications().m_tip_block != current_tip && notifications().m_tip_block != uint256::ZERO) || chainman().m_interrupt;
+                // We need to wait for m_tip_block to be set AND for the value
+                // to be different than the current_tip value.
```

Comment:
nit: https://languagetool.org complains here
```
Did you mean 'different from'? 'Different than' is often considered colloquial style. 
Incorrect:  Are human beings any different than animals? 
Correct:  Are human beings any different from animals? 
```

```suggestion
                // We need to wait for m_tip_block to be set AND for the value
                // to differ from the current_tip value.
```

Reply:
Taken